rele/subscription.py

```diff
@@ -178,8 +180,29 @@ def sub_process_landscape_photos(data, **kwargs):
                       the sub regarding their attributes.
     :return: :class:`~rele.subscription.Subscription`
     """
+    if filter_by and not (
+        callable(filter_by)
+        or (
+            isinstance(filter_by, Iterable)
+            and all(callable(filter) for filter in filter_by)
+        )
+    ):
+        raise ValueError("Filter_by must be a callable or a list of callables.")
 
     def decorator(func):
+        args_spec = getfullargspec(func)
+        if len(args_spec.args) != 1:
```

Comment:
I think this should be `< 2`. We will always be passing atleast two args into the callback. One is the `data` and other(s) will be the attributes. One of them should always be present which is the `published_at` field.

Reply:
Yeah but here is checking the arguments, not the keyword arguments that are optional.